discovery/manager_test.go

```diff
@@ -774,6 +774,47 @@ scrape_configs:
 	verifyPresence(discoveryManager.targets, poolKey{setName: "prometheus", provider: "static/0"}, "{__address__=\"bar:9090\"}", false)
 }
 
+func TestProvidersFromConfigDoesNotModifyStaticProviderTargets(t *testing.T) {
+	cfgText := `
+scrape_configs:
+ - job_name: 'prometheus'
+   static_configs:
+   - targets: ["foo:9090"]
+   - targets: ["bar:9090"]
+`
+	originalConfig := &config.Config{}
+	if err := yaml.UnmarshalStrict([]byte(cfgText), originalConfig); err != nil {
+		t.Fatalf("Unable to load YAML config cfgYaml: %s", err)
+	}
+
+	processedConfig := &config.Config{}
+	if err := yaml.UnmarshalStrict([]byte(cfgText), processedConfig); err != nil {
+		t.Fatalf("Unable to load YAML config cfgYaml: %s", err)
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	discoveryManager := NewManager(ctx, nil)
+	go discoveryManager.Run()
+
+	for i, scrpConfig := range processedConfig.ScrapeConfigs {
+		origScrpCfg := originalConfig.ScrapeConfigs[i]
+		for _, discoverer := range discoveryManager.providersFromConfig(scrpConfig.ServiceDiscoveryConfig) {
```

Comment:
I would find it easier to read if the test called `discoveryManager.ApplyConfig()` then waited on `<- discoveryManager.SyncCh()` as it is done in the other tests.

Reply:
Yeah, I agree that's a better way to do it.  Branch is now updated, thanks!